### PR TITLE
REST API support

### PR DESCRIPTION
**Pull request by slint**

### CODE REVIEW

**Comment by slint** [comment]
Copy of the accidentally merged #197

**Comment by zzacharo** [comment]
Needs to resolve this https://github.com/inveniosoftware/invenio-oauthclient/issues/201 so tests can pass!

File: invenio_oauthclient/views/rest.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
```
Comment (lnielsen) [inline invenio_oauthclient/views/rest.py]: Ditto here, regarding code duplication.

**Comment by lnielsen** [comment]
For the upcoming sprint, see my comments. Please ping me, once ready to have it rereviewed.

File: invenio_oauthclient/contrib/cern.py [hunk 2]
```diff
@@ -159,6 +219,44 @@
...
+    ),
+    logout_url='https://login.cern.ch/adfs/ls/?wa=wsignout1.0',
+    authorized_redirect_url='/',
+    disconnect_redirect_url='/',
+    signup_redirect_url='/',
+    error_redirect_url='/',
+    params=dict(
+        base_url='https://oauth.web.cern.ch/',
+        request_token_url=None,
+        access_token_url='https://oauth.web.cern.ch/OAuth/Token',
```
[the remark this answers is not in the document]
Comment (zzacharo) [reply invenio_oauthclient/contrib/cern.py:248]: Old one :)

File: invenio_oauthclient/utils.py [hunk 3]
```diff
@@ -105,11 +126,7 @@ def oauth_register(form):
     """
     if form.validate():
         data = form.to_dict()
-        if not data.get('password'):
```
[the remark this answers is not in the document]
Comment (zzacharo) [reply invenio_oauthclient/utils.py:108]: It is transferred https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/utils.py#L154

**Comment by zzacharo** [comment]
Deprecated by #205

**Comment by ppanero** [comment]
Closed in favour of #205

**Pull request closed without merging**